puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_proxy_develop_pr_core.yaml

```diff
@@ -2,44 +2,47 @@
     name: test_proxy_develop_pr_core
     description: |
       Tests a Smart Proxy pull request under each Ruby and Puppet configuration.
+    project-type: matrix
     concurrent: true
     logrotate:
       daysToKeep: 45
       numToKeep: -1
     properties:
       - github:
           url: https://github.com/theforeman/smart-proxy/
-    parameters:
-      - pr_parameters
-      - string:
-          name: branch
-          default: develop
-          description: |
-            Base branch to merge PR into
     scm:
       - git:
           url: https://github.com/theforeman/smart-proxy/
           wipe-workspace: true
           prune: true
           branches:
-            - 'project/${branch}'
+            - '${ghprbActualCommit}'
+          refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*'
+    triggers:
+      - github_pr:
+          context: core
     axes:
       - axis:
           type: user-defined
           name: ruby
           values:
-            - 2.0.0
-            - 2.1
-            - 2.2
-            - 2.3
+            - '2.0.0'
+            - '2.1'
+            - '2.2'
+            - '2.3'
+            - '2.4'
+            - '2.5'
       - axis:
           type: user-defined
           name: puppet
           values:
-            - 3.4.0
-            - 3.8.0
-            - 4.2.0
-            - 4.4.0
+            - '3.4.0'
+            - '3.8.0'
+            - '4.2.0'
+            - '4.4.0'
+            - '4.10.0'
+            - '5.0'
```

Comment:
I'd just test 5.0 to keep the matrix minimal. That should resolve to the latest 5.x.